aws/table_aws_wellarchitected_milestone.go

```diff
@@ -0,0 +1,154 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/wellarchitected"
+	"github.com/aws/aws-sdk-go-v2/service/wellarchitected/types"
+
+	wellarchitectedv1 "github.com/aws/aws-sdk-go/service/wellarchitected"
+
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsWellArchitectedMilestone(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_wellarchitected_milestone",
+		Description: "AWS Well-Architected Milestone",
+		Get: &plugin.GetConfig{
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "milestone_number", Require: plugin.Required},
+				{Name: "workload_id", Require: plugin.Required},
+			},
+			Hydrate: getWellArchitectedMilestone,
+		},
+		List: &plugin.ListConfig{
+			ParentHydrate: listWellArchitectedWorkloads,
+			Hydrate:       listWellArchitectedMilestones,
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "workload_id", Require: plugin.Optional},
+			},
+		},
+		GetMatrixItemFunc: SupportedRegionMatrix(wellarchitectedv1.EndpointsID),
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "milestone_name",
+				Description: "The name of the milestone in a workload. Milestone names must be unique within a workload.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "milestone_number",
+				Description: "The milestone number. A workload can have a maximum of 100 milestones.",
+				Type:        proto.ColumnType_INT,
+			},
+			{
+				Name:        "recorded_at",
+				Description: "The date and time recorded.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
```

Comment:
@karanpopat why didn't we add `workload_summary` column as per https://docs.aws.amazon.com/wellarchitected/latest/APIReference/API_ListMilestones.html